### ISSUE TITLE
Remove USE_SHARED_ flags to avoid issues with Linux distros

### ISSUE THREAD

**Issue by dacap** [aseprite/aseprite#2843]
Our preferred distribution mechanism is linked everything statically, and it looks like using `USE_SHARED_*` flags can cause some problems in different issues. We would prefer to remove all those flags.

**Comment by mohad12211** [comment]
I agree with this actually, dynamic linking might save a few Megabytes in the disk and a few minutes in compiling, but causes problems especially between linux distributions that would take much more time to solve.

**Comment by ryandesign** [comment]
What you prefer is not necessarily the same as what your users prefer.

Please do not remove the USE_SHARED options. Please add additional USE_SHARED options as has been requested in existing PRs and issues.

If I already have various libraries installed, I want Aseprite to use them; I do not want to have to compile them again. Maybe the versions I have are newer. Maybe I have fixed bugs in the versions I have, and I don't want to have to fix those same bugs in your bundled versions. For example, you may recall that some time ago I reported that I had successfully built Aseprite on an Apple Silicon Mac, which was possible because I was using newer libraries than the ones you bundled.

**Comment by ISSOtm** [comment]
Please do not remove the `USE_SHARED_` flags. Yes, they can cause incompatibilities, but this is an usual part of the build process; statically linking everything is a good enough default for users not savvy enough to recognize them, but more "advanced" users know what they are doing, and prefer the option to be available.

I am currently personally having an issue with the bundled `libwebp`, for example, where `cmake --install` also attempts to install its compiled components. (Also, for some reason, I must enable either `gif2webp` or `img2webp` for one of Aseprite's required libs to be compiled.) I can work around it, but using a shared lib instead would get rid of the issue.

Other than that, my points are the same as @ryandesign's. I'd also enjoy more of those flags, personally.

If the compatibility issues aren't obvious, maybe a line could be added in the README next to the mention of `USE_SHARED_*` pointing them out.

To ease solving these issues, is there a relatively easy or convenient way to check which versions of its depended libs Aseprite expects?
